testsuite/oiiotool/run.py

```diff
@@ -256,8 +256,8 @@
 # test expression substitution
 command += oiiotool ("tahoe-small.tif --pattern fill:top=0,0,0,0:bottom=0,0,1,1 " +
                      "{TOP.geom} {TOP.nchannels} -d uint8 -o exprgradient.tif")
-command += oiiotool ("tahoe-small.tif -cut {TOP.width-40}x{TOP.height-40}+{TOP.x+20}+{TOP.y+20} -d uint8 -o exprcropped.tif")
-
+command += oiiotool ("tahoe-small.tif -cut '{TOP.width-20* 2}x{TOP.height-40+(4*2- 2 ) /6-1}+{TOP.x+100.5-80.5 }+{TOP.y+20}' -d uint8 -o exprcropped.tif")
+command += oiiotool ("tahoe-small.tif -o '{\"exprstrcat\" + TOP.compression + \".tif\"}'")
```

Comment:
I understand that you are just testing string concatenation, but I think any sane person would find this more clear:

    oiiotool tahoe-small.tif -o exprstrcat{TOP.compression}.tif

I'm not necessarily arguing against '+' for string concatenation, but I am unclear when one would prefer that notation, and whether supporting it that adds any unnecessarily complexity to the expression implementation.